applications/GeoMechanicsApplication/custom_elements/small_strain_U_Pw_diff_order_element.hpp

```diff
@@ -11,54 +11,43 @@
 //                   Vahid Galavi
 //
 
-#if !defined(KRATOS_GEO_SMALL_STRAIN_U_PW_DIFF_ORDER_ELEMENT_H_INCLUDED)
-#define KRATOS_GEO_SMALL_STRAIN_U_PW_DIFF_ORDER_ELEMENT_H_INCLUDED
-
-#include "custom_retention/retention_law.h"
-#include "geometries/geometry_data.h"
-#include "includes/constitutive_law.h"
-#include "includes/define.h"
-#include "includes/element.h"
-#include "includes/kratos_export_api.h"
-#include "includes/serializer.h"
-#include "includes/smart_pointers.h"
-#include "includes/ublas_interface.h"
+#pragma once
 
-#include <iosfwd>
-#include <memory>
-#include <string>
-#include <vector>
+#include "custom_elements/U_Pw_base_element.hpp"
+#include "includes/kratos_export_api.h"
 
 namespace Kratos
 {
 
-class StressStatePolicy;
-
-template <class T, std::size_t N>
-class array_1d;
-
-class KRATOS_API(GEO_MECHANICS_APPLICATION) SmallStrainUPwDiffOrderElement : public Element
+class KRATOS_API(GEO_MECHANICS_APPLICATION) SmallStrainUPwDiffOrderElement : public UPwBaseElement
 {
 public:
     KRATOS_CLASS_INTRUSIVE_POINTER_DEFINITION(SmallStrainUPwDiffOrderElement);
 
-    //----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
-    // Default constructor
-    SmallStrainUPwDiffOrderElement();
+    using UPwBaseElement::mConstitutiveLawVector;
+    using UPwBaseElement::mIsInitialised;
+    using UPwBaseElement::mRetentionLawVector;
+    using UPwBaseElement::mStateVariablesFinalized;
+    using UPwBaseElement::mStressVector;
+
+    using UPwBaseElement::UPwBaseElement;
 
-    // Constructor 1
     SmallStrainUPwDiffOrderElement(IndexType                          NewId,
                                    GeometryType::Pointer              pGeometry,
-                                   std::unique_ptr<StressStatePolicy> pStressStatePolicy);
+                                   std::unique_ptr<StressStatePolicy> pStressStatePolicy)
+        : UPwBaseElement(NewId, pGeometry, std::move(pStressStatePolicy))
+    {
+    }
```

Comment:
I'm having difficulty to understand why these constructor implementations have moved to the header.

Reply:
UPwBaseElement and its children have constructors in headers. Try to follow the same style.

Reply:
Okay, I think one of the reasons the UPwBaseElement had these constructors in the headers, is that it was a template class before. I think for the diff order element it's not needed. We can maybe have a short discussion with @avdg81, I don't think it should block this PR if the pipelines are green, but it could be taken up in the next, since that's in the same class.